bazel/boringssl_static.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/BUILD b/BUILD
+index d7c731bf6..315cdeca0 100644
+--- a/BUILD
++++ b/BUILD
+@@ -88,6 +88,7 @@ boringssl_copts = select({
+     ":windows_x86_64": [
+         "-DWIN32_LEAN_AND_MEAN",
+         "-DOPENSSL_NO_ASM",
++        "-DBORINGSSL_IMPLEMENTATION",
```

Comment:
what is this for?

Reply:
The BoringSSL source distinguishes source compilation from library consumption with this macro, generally a no-op on linux, but significant on Windows. Here above, we are building the library objects.
See bazel-envoy\external\boringssl\src\include\openssl\base.h

Reply:
@lizan did you have any follow-up concerns on this specific correction?

Reply:
No. cc @davidben @agl this should go to boringssl directly.